### DISCUSSION TITLE
I have a project template for C3, what next?

### DISCUSSION THREAD

**Discussion by lerno** [aduros/wasm4#607] [General]
I've created a WASM4 project template for C3. In C3, the normal thing would be to use the built in build file / project, so that's what I did. However, it's also possible to create a make file and run from the command line. OR build the project from the make file.

So my question is, if I want to contribute this template – how do I go about it? What is the preferred format?

**Comment by SLiV9** [comment]
Oh cool C3! My [pull request for Penne](https://github.com/aduros/wasm4/pull/606) just got merged so you could take inspiration from that. I used a Makefile but I don't think you need to; Zig calls `zig build` directly and AssemblyScript uses `npm`. As long as you have a one-liner to build a cart.wasm, you can add support for *w4 new* and *w4 watch* [here](https://github.com/aduros/wasm4/blob/35fb9facbee2159dde3881e64a61710bae8db828/cli/lib/new.js) and [here](https://github.com/aduros/wasm4/blob/35fb9facbee2159dde3881e64a61710bae8db828/cli/lib/watch.js).

But maybe there is a preference for `make`, to keep the documentation simple/uniform?

**Reply by lerno** [reply]
I actually followed your edits for Penne to find all the places where I had to add C3 examples. Very helpful!
